src/pudl/etl.py

```diff
@@ -446,6 +446,76 @@ def _read_static_pudl_tables() -> dict[str, pd.DataFrame]:
     return {"political_subdivisions": POLITICAL_SUBDIVISIONS}
 
 
+def make_datasources_table(datasets_settings, ds):
```

Comment:
I wonder if it makes sense for this function to be a method of the DatasetsSettings class, given it's mostly converting the DatasetsSettings to a new data structure. I don't think it really matters, though.

Reply:
hm this is a good idea! let me think about if this will me easy to slurp over there.